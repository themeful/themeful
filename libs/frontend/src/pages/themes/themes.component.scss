.design-tokens {
  display: table;

  &__row {
    display: table-row;
    font-size: 12px;

    &:nth-child(even) {
      background-color: rgba(200, 200, 200, 0.1);
    }

    &:nth-child(odd) {
      background-color: rgba(50, 50, 50, 0.1);
    }

    & > * {
      vertical-align: top;
      display: table-cell;
      padding: 12px;
    }

    &--inline {
      margin-bottom: 8px;
      display: flex;
      align-items: center;
      gap: 4px;
    }
  }

  &__header > * {
    padding: 0;
  }

  &__value {
    border: 1px solid rgba(200, 200, 200, 0.55);
    border-radius: 8px;
  }

  &__media-label {
    margin-top: 4px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    max-width: 200px;
  }

  &__add-theme,
  &__add-design-token {
    vertical-align: middle;
    margin-left: 8px;
  }

  &__name {
    margin: 0 0 8px;
  }

  &__desc {
    margin-bottom: 8px;
  }

  &__theme-value {
    text-align: center;
  }

  &__add-value {
    margin-top: 16px;
  }

  &__select-aliasTokens {
    text-align: center;
    padding-top: 16px;
  }

  ul {
    list-style: none;
    padding: 0;
    margin: 0;
  }
}
