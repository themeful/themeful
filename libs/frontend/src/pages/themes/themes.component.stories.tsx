import { h } from '@stencil/core'

export default {
  title: 'Pages/Themes',
  args: {
    first: 'John',
    middle: 'S',
    last: 'Doe',
  },
}

export const themes = (args) => {
  return (
    <div>
      <div class="header">
        <h1>Themes</h1>
      </div>
      <div class="content">
        <div style={{ position: 'relative' }}>
          <tf-themes {...args} />
        </div>
      </div>
    </div>
  )
}
