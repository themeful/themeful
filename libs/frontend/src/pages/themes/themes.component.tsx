import { Component, Fragment, h, Host, Prop, State } from '@stencil/core'
import {
  AliasTokens,
  APIBundle,
  DesignTokenRow,
  DesignTokens,
  Dt2At,
  ExtendedValueDetail,
  ExtendedValueDetails,
  StyleGuides,
  StyleMap,
  Theme,
  Themes,
} from '@typings'
import { Observable, Subscription } from 'rxjs'
// import { ThemeService } from '../..'

@Component({
  tag: 'tf-themes',
  styleUrl: 'themes.component.scss',
  shadow: true,
})
export class ThemesComponent {
  /** Style Guides */
  @Prop() bundle$: Observable<APIBundle>

  @State() rows: DesignTokenRow[]
  @State() themeNames: any
  private themes: Themes
  private designTokens: DesignTokens
  private aliasTokens: AliasTokens
  private styleMap: StyleMap
  private sgBases: { name: string; key: string }[]
  private dt2at: Dt2At = {}
  private groups: string[]

  private sub?: Subscription

  public componentWillLoad(): void {
    this.sub = this.bundle$?.subscribe(([styleGuides, designTokens, aliasTokens, themes]) => {
      this.themes = themes
      this.aliasTokens = aliasTokens
      this.designTokens = designTokens
      this.groups = []
      const { styleMap, sgNames } = this.transformStyleGuides(styleGuides)
      this.styleMap = styleMap
      this.sgBases = Object.keys(sgNames).map((key) => {
        return { name: sgNames[key], key }
      })
      const themeNames = Object.keys(themes).map((key) => ({
        displayName: `${sgNames[themes[key].styleGuide]} - ${themes[key].name}`,
        styleGuide: themes[key].styleGuide,
        name: themes[key].name,
        key,
      }))

      const rows: DesignTokenRow[] = []
      Object.keys(designTokens).forEach((designToken) => {
        this.dt2at[designToken] = designTokens[designToken].aliasTokens
        if (!this.groups.includes(designTokens[designToken].group)) {
          this.groups.push(designTokens[designToken].group)
        }
        const themeValues: ExtendedValueDetails[] = []
        Object.values(themes).forEach((theme: Theme) => {
          const themeValue: ExtendedValueDetails = []
          if (theme.styles[designToken]) {
            Object.entries(theme.styles[designToken]).forEach(([media, { style, direct }]) => {
              const themeMedia: ExtendedValueDetail = {
                media,
                name: styleMap[media] ? styleMap[media].name : 'Default',
                global: styleMap[media] ? styleMap[media].global : false,
                style: style ? { ...styleMap[style], key: style } : undefined,
                direct,
              }
              themeValue.push(themeMedia)
            })
          }
          themeValues.push(themeValue)
        })
        rows.push({ ...designTokens[designToken], token: designToken, themeValues })
      })
      this.rows = rows
      this.themeNames = themeNames
    })
  }

  private transformStyleGuides(styleGuides: StyleGuides) {
    const sgNames: { [styleguide: string]: string } = {}
    const styleMap: StyleMap = {}
    Object.keys(styleGuides).forEach((styleguide) => {
      sgNames[styleguide] = styleGuides[styleguide].name
      Object.keys(styleGuides[styleguide].styles).forEach((valueKey) => {
        styleMap[`${styleguide}_${valueKey}`] = {
          ...styleGuides[styleguide].styles[valueKey],
          global: styleguide === 'global',
        }
      })
    })
    return { styleMap, sgNames }
  }

  public disconnectedCallback(): void {
    this.sub?.unsubscribe()
  }

  private openThemeForm = (theme?: string): void => {
    console.log('openThemeForm', theme, this.themes, this.sgBases)
    // this.formData$.next({})
  }

  private openDesignTokenForm = (data?: any): void => {
    console.log('openDesignTokenForm', data)
    // this.formData$.next({})
  }

  private openAliasTokenSelect = (designToken: string): void => {
    console.log('openAliasTokenSelect', designToken, this.designTokens, this.aliasTokens)
    // this.formData$.next({})
  }
  private openThemeValueForm = (data: any): void => {
    console.log('openThemeValueForm', data, this.styleMap)
    // this.formData$.next({})
  }

  private rescanAliasTokens = (): void => {
    // this.formData$.next({})
  }

  private renderHeader(): HTMLElement {
    return (
      <div class="design-tokens__row design-tokens__header">
        <div>
          <h3>Design Token</h3>
        </div>
        <div>
          <h3>Alias Tokens</h3>
        </div>
        {this.themeNames.map((theme) => (
          <div>
            <h3>
              {theme.displayName}
              <tf-button
                {...{
                  onClick: () => this.openThemeForm(theme.key),
                  title: 'edit theme',
                  size: 'icon',
                }}
              >
                <tf-icon icon="pen" />
              </tf-button>
            </h3>
          </div>
        ))}
      </div>
    )
  }

  private renderDesignTokenRow(row): HTMLElement {
    return (
      <div class="design-tokens__row">
        <div>
          <h5 class="design-tokens__name">
            {row.name}
            <tf-button
              {...{
                onClick: () => this.openDesignTokenForm({ designToken: row }),
                title: 'edit design token',
                size: 'icon',
              }}
            >
              <tf-icon icon="pen" />
            </tf-button>
          </h5>
          <div class="design-tokens__desc">{row.description}</div>
          {row.short && (
            <div>
              <pre>({row.token})</pre>
              <pre>Using: --{row.short}</pre>
            </div>
          )}
          {!row.short && <pre>--{row.token}</pre>}
        </div>
        {row.aliasTokens.length > 0 && (
          <ul>
            {row.aliasTokens.map((aliasToken) => (
              <li>
                <pre>${aliasToken}</pre>
              </li>
            ))}
            <li class="design-tokens__select-aliasTokens">
              <tf-button
                {...{
                  onClick: () => this.openAliasTokenSelect(row.token),
                  title: 'select alias tokens',
                }}
              >
                Edit Alias Tokens
              </tf-button>
            </li>
          </ul>
        )}

        {row.aliasTokens.length === 0 && (
          <div>
            <tf-button
              {...{
                onClick: () => this.openAliasTokenSelect(row.token),
                title: 'select alias tokens',
              }}
            >
              Add Alias Tokens
            </tf-button>
          </div>
        )}
        {row.themeValues.map((themeValue) => (
          <div class="design-tokens__theme-value">
            {themeValue.map((themeMedia) => (
              <Fragment>
                {(themeValue.length > 1 || themeMedia.media !== 'default') && (
                  <span class="design-tokens__media-label icon-text-wrapper">
                    {themeMedia.global && <tf-icon size="small" icon="globe" />}
                    {themeMedia.name}
                  </span>
                )}

                <tf-property
                  {...{
                    extendedStyle: themeMedia.style ?? themeMedia.direct,
                    onEdit: () => this.openThemeValueForm({ designToken: row.token, themeMedia }), //{themeMedias, themeMedia, token: row.token, themeIndex: i}
                  }}
                  class="design-tokens__value"
                ></tf-property>
              </Fragment>
            ))}

            <tf-button
              {...{
                onClick: () => this.openThemeValueForm({ designToken: row.token }), //{themeMedias, token: row.token, themeIndex: i}
                title: 'add value',
                class: 'design-tokens__add-value',
              }}
            >
              Add Value
            </tf-button>
          </div>
        ))}
      </div>
    )
  }

  public render(): HTMLTfThemesElement {
    return (
      <Host>
        <nav>
          <tf-button {...{ onClick: () => this.openThemeForm(), title: 'add theme' }}>
            Add Theme
          </tf-button>
          <tf-button {...{ onClick: () => this.openDesignTokenForm(), title: 'add design token' }}>
            Add Design Token
          </tf-button>
          <tf-button {...{ onClick: () => this.rescanAliasTokens(), title: 'rescan alias tokens' }}>
            Rescan Alias Tokens
          </tf-button>
        </nav>
        <main class="design-tokens">
          {this.themeNames && this.renderHeader()}
          {this.rows && this.rows.map((row) => this.renderDesignTokenRow(row))}
        </main>
        {/* <tf-form-integration {...{ formData$: this.formData$, onAction: this.onAction }} /> */}
      </Host>
    )
  }
}
