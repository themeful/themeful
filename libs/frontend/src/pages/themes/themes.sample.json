[
  {
    "global": {
      "name": "Global",
      "baseFontSize": 16,
      "styles": {
        "base_black": {
          "name": "Black",
          "type": "color",
          "group": "base",
          "value": "#333333"
        },
        "base_lIght": {
          "name": "LIght",
          "type": "color",
          "group": "base",
          "value": "#ffffff"
        },
        "gray_100": {
          "name": "100",
          "type": "color",
          "group": "gray",
          "value": "rgba(0, 0, 0, 0.75)"
        },
        "gray_200": {
          "name": "200",
          "type": "color",
          "group": "gray",
          "value": "rgba(0, 0, 0, 0.60)"
        },
        "gray_300": {
          "name": "300",
          "type": "color",
          "group": "gray",
          "value": "rgba(0, 0, 0, 0.45)"
        },
        "gray_400": {
          "name": "400",
          "type": "color",
          "group": "gray",
          "value": "rgba(0, 0, 0, 0.30)"
        },
        "gray_500": {
          "name": "500",
          "type": "color",
          "group": "gray",
          "value": "rgba(0, 0, 0, 0.15)"
        },
        "space_tiny": {
          "type": "size",
          "group": "space",
          "name": "Tiny",
          "value": "4px"
        },
        "space_small": {
          "type": "size",
          "group": "space",
          "name": "Small",
          "value": "8px"
        },
        "space_smallMedium": {
          "name": "Small Medium",
          "type": "size",
          "group": "space",
          "value": "0.75rem"
        },
        "space_medium": {
          "type": "size",
          "group": "space",
          "name": "Medium",
          "value": "16px"
        },
        "space_large": {
          "type": "size",
          "group": "space",
          "name": "Large",
          "value": "24px"
        },
        "space_xlarge": {
          "type": "size",
          "group": "space",
          "name": "X-Large",
          "value": "32px"
        },
        "mediaQuery_aboveSmallMobile": {
          "name": "Above Small Mobile",
          "type": "mediaquery",
          "group": "mediaQuery",
          "value": "screen and (min-width: 321px)"
        },
        "mediaQuery_aboveMobile": {
          "name": "Above Mobile",
          "type": "mediaquery",
          "group": "mediaQuery",
          "value": "screen and (min-width: 664px)"
        },
        "mediaQuery_aboveTablet": {
          "name": "Above Tablet",
          "type": "mediaquery",
          "group": "mediaQuery",
          "value": "screen and (min-width: 992px)"
        },
        "mediaQuery_aboveDesktop": {
          "name": "Above Desktop",
          "type": "mediaquery",
          "group": "mediaQuery",
          "value": "screen and (min-width: 1200px)"
        }
      }
    },
    "styleGuide1": {
      "name": "StyleGuide 1",
      "baseFontSize": 16,
      "styles": {
        "action_primary": {
          "name": "Primary",
          "type": "color",
          "group": "action",
          "value": "#31ed31"
        },
        "action_secondary": {
          "name": "Secondary",
          "type": "color",
          "group": "action",
          "value": "#2ec22e"
        },
        "action_tertiary": {
          "name": "Tertiary",
          "type": "color",
          "group": "action",
          "value": "#1e961e"
        },
        "brand_primary": {
          "type": "color",
          "group": "brand",
          "name": "Primary",
          "value": "#ff0000"
        },
        "brand_secondary": {
          "name": "Secondary",
          "type": "color",
          "group": "brand",
          "value": "#d42828"
        },
        "brand_tertiary": {
          "name": "Tertiary",
          "type": "color",
          "group": "brand",
          "value": "#af2323"
        },
        "font_primary": {
          "name": "Primary",
          "type": "font",
          "group": "font",
          "value": "Verdana, Geneva, Tahoma, sans-serif"
        },
        "content_primary": {
          "type": "font-size",
          "group": "content",
          "name": "Primary",
          "value": "1.2rem"
        },
        "mediaQuery_aboveSmallDesktop": {
          "name": "Above Desktop",
          "type": "mediaquery",
          "group": "mediaQuery",
          "value": "screen and (min-width: 1100px)"
        }
      }
    },
    "styleGuide2": {
      "name": "StyleGuide 2",
      "baseFontSize": 16,
      "styles": {
        "action_primary": {
          "name": "Primary",
          "type": "color",
          "group": "action",
          "value": "#ff5555"
        },
        "action_secondary": {
          "name": "Secondary",
          "type": "color",
          "group": "action",
          "value": "#d22828"
        },
        "action_tertiary": {
          "name": "Tertiary",
          "type": "color",
          "group": "action",
          "value": "#9f1c1c"
        },
        "brand_primary": {
          "name": "Primary",
          "type": "color",
          "group": "brand",
          "value": "#577bf1"
        },
        "brand_secondary": {
          "name": "Secondary",
          "type": "color",
          "group": "brand",
          "value": "#4646ed"
        },
        "brand_tertiary": {
          "name": "Tertiary",
          "type": "color",
          "group": "brand",
          "value": "#3f55b1"
        },
        "font_primary": {
          "name": "Primary",
          "type": "font",
          "group": "font",
          "value": "'Times New Roman', Times, serif"
        },
        "content_normal": {
          "type": "font-size",
          "group": "content",
          "name": "Normal",
          "value": "0.9rem"
        }
      }
    },
    "styleGuide3": {
      "name": "StyleGuide 3",
      "baseFontSize": 10,
      "styles": {
        "base_primary": {
          "name": "Primary",
          "type": "color",
          "group": "base",
          "value": "#2be053"
        }
      }
    }
  },
  {
    "dtFontColorPrimary": {
      "name": "Normal Font Color",
      "type": "color",
      "group": "content",
      "description": "Font color for normal text",
      "properties": ["color"],
      "aliasTokens": ["atBaseFontColor", "atButtonFontColor"]
    },
    "dtFontSize100": {
      "name": "Normal Font Size",
      "type": "font-size",
      "group": "content",
      "description": "Font size for normal text",
      "properties": ["font-size"],
      "aliasTokens": ["atButtonFontSize"]
    },
    "dtPrimaryFont": {
      "name": "Primary Font",
      "type": "font",
      "group": "content",
      "description": "Primary Font",
      "properties": [],
      "aliasTokens": []
    },
    "dtActionBg": {
      "name": "Action Background",
      "type": "color",
      "group": "controls",
      "description": "Background for action elements",
      "properties": ["color", "background-color"],
      "aliasTokens": ["atButtonBackground"]
    },
    "dtActionBgHover": {
      "name": "Action Background Hover",
      "type": "color",
      "group": "controls",
      "description": "Hover Background for action elements",
      "properties": [],
      "aliasTokens": ["atButtonBackgroundHover"]
    },
    "dtControlBorderRadius": {
      "name": "Default Border Radius",
      "type": "size",
      "group": "controls",
      "description": "Control Border Radius",
      "properties": ["border-radius"],
      "aliasTokens": ["atButtonBorderRadius"]
    }
  },
  {
    "atBaseFontColor": {
      "component": null,
      "files": null,
      "properties": ["color"],
      "extern": true,
      "crawled": false
    },
    "atButtonBackground": {
      "component": ["Button"],
      "files": ["libs/components/src/lib/button/button.component.scss"],
      "properties": ["background-color"],
      "extern": true,
      "crawled": true
    },
    "atButtonBackgroundHover": {
      "component": ["Button"],
      "files": ["sample/components/button/button.component.scss"],
      "properties": ["background-color"],
      "extern": false,
      "crawled": true
    },
    "atButtonBorderRadius": {
      "component": ["Button"],
      "files": ["sample/components/button/button.component.scss"],
      "properties": ["border-radius"],
      "extern": false,
      "crawled": true
    },
    "atButtonFontColor": {
      "component": ["Button"],
      "files": ["sample/components/button/button.component.scss"],
      "properties": ["color"],
      "extern": false,
      "crawled": true
    },
    "atButtonFontSize": {
      "component": ["Button"],
      "files": ["sample/components/button/button.component.scss"],
      "properties": ["font-size"],
      "extern": false,
      "crawled": true
    },
    "atButtonHorizontalSpace": {
      "component": ["Button"],
      "files": ["sample/components/button/button.component.scss"],
      "properties": ["padding"],
      "extern": false,
      "crawled": true
    },
    "atButtonVerticalSpace": {
      "component": ["Button"],
      "files": ["sample/components/button/button.component.scss"],
      "properties": ["padding"],
      "extern": false,
      "crawled": true
    },
    "atCardBackground": {
      "component": ["Card"],
      "files": ["sample/components/card/card.component.scss"],
      "properties": ["background-color"],
      "extern": false,
      "crawled": true
    }
  },
  {
    "styleGuide1_dark": {
      "name": "Dark",
      "styleGuide": "styleGuide1",
      "styles": {
        "dtActionBg": {
          "default": {
            "style": "styleGuide1_action_tertiary"
          }
        },
        "dtActionBgHover": {
          "default": {
            "style": "styleGuide1_action_secondary"
          }
        },
        "dtControlBorderRadius": {
          "default": {
            "style": "global_space_tiny"
          }
        },
        "dtFontColorPrimary": {
          "default": {
            "style": "global_base_lIght"
          }
        },
        "dtFontSize100": {
          "default": {
            "style": "styleGuide1_content_primary"
          },
          "styleGuide1_mediaQuery_aboveSmallDesktop": {
            "directValue": {
              "value": "13px",
              "type": "font-size"
            }
          }
        },
        "dtPrimaryFont": {
          "default": {
            "style": "styleGuide1_font_primary"
          }
        }
      }
    },
    "styleGuide1_light": {
      "name": "Light",
      "styleGuide": "styleGuide1",
      "styles": {
        "dtActionBg": {
          "default": {
            "style": "styleGuide1_action_primary"
          }
        },
        "dtActionBgHover": {
          "default": {
            "style": "styleGuide1_action_secondary"
          }
        },
        "dtControlBorderRadius": {
          "default": {
            "style": "global_space_small"
          }
        },
        "dtFontColorPrimary": {
          "default": {
            "style": "global_base_black"
          }
        },
        "dtFontSize100": {
          "default": {
            "style": "styleGuide1_content_primary"
          }
        },
        "dtPrimaryFont": {
          "default": {
            "style": "styleGuide1_font_primary"
          }
        }
      }
    },
    "styleGuide2_dark": {
      "name": "Dark",
      "styleGuide": "styleGuide2",
      "styles": {
        "dtActionBg": {
          "default": {
            "style": "styleGuide2_brand_tertiary"
          }
        },
        "dtActionBgHover": {
          "default": {
            "style": "styleGuide2_brand_secondary"
          }
        },
        "dtControlBorderRadius": {
          "default": {
            "style": "global_space_small"
          }
        },
        "dtFontColorPrimary": {
          "default": {
            "style": "global_base_lIght"
          }
        },
        "dtFontSize100": {
          "default": {
            "style": "styleGuide2_content_normal"
          }
        },
        "dtPrimaryFont": {
          "default": {
            "style": "styleGuide2_font_primary"
          }
        }
      }
    },
    "styleGuide2_light": {
      "name": "Light",
      "styleGuide": "styleGuide2",
      "styles": {
        "dtActionBg": {
          "default": {
            "style": "styleGuide2_brand_primary"
          }
        },
        "dtActionBgHover": {
          "default": {
            "style": "styleGuide2_brand_secondary"
          }
        },
        "dtControlBorderRadius": {
          "default": {
            "style": "global_space_tiny"
          }
        },
        "dtFontColorPrimary": {
          "default": {
            "style": "global_base_black"
          }
        },
        "dtFontSize100": {
          "default": {
            "style": "styleGuide2_content_normal"
          }
        },
        "dtPrimaryFont": {
          "default": {
            "style": "styleGuide2_font_primary"
          }
        }
      }
    },
    "styleGuide3_default": {
      "name": "Default",
      "styleGuide": "styleGuide3",
      "styles": {
        "dtActionBg": {
          "default": {
            "directValue": {
              "value": "#aaaaaa",
              "type": "color"
            }
          },
          "global_mediaQuery_aboveSmallMobile": {
            "directValue": {
              "value": "rgba(180, 180, 180, 0.8)",
              "type": "color"
            }
          }
        },
        "dtActionBgHover": {
          "default": {
            "style": "styleGuide3_base_primary"
          }
        },
        "dtControlBorderRadius": {
          "default": {
            "style": "global_space_tiny"
          },
          "global_mediaQuery_aboveSmallMobile": {
            "style": "global_space_small"
          },
          "global_mediaQuery_aboveMobile": {
            "style": "global_space_medium"
          },
          "global_mediaQuery_aboveTablet": {
            "style": "global_space_large"
          }
        },
        "dtFontColorPrimary": {
          "default": {
            "style": "styleGuide3_base_primary"
          }
        },
        "dtFontSize100": {
          "global_mediaQuery_aboveSmallMobile": {
            "directValue": {
              "type": "font-size",
              "value": "1rem"
            }
          },
          "global_mediaQuery_aboveMobile": {
            "directValue": {
              "type": "font-size",
              "value": "2rem"
            }
          }
        }
      }
    }
  }
]
