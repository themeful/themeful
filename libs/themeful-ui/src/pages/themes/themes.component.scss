:host {
  width: 100%;
  height: 100%;
}

.design-tokens {
  font: var(--tf-default-font);
  border-collapse: collapse;

  &__wrapper {
    width: 100%;
    height: 100%;
    overflow: auto;
  }

  &__row {
    font-size: 0.75rem;

    &:nth-child(odd) {
      background-color: rgba(50, 50, 50, 0.1);
    }

    &:nth-child(even),
    &:first-child {
      background-color: rgba(200, 200, 200, 0.1);
    }

    // &:first-child {
    //   position: sticky;
    //   top: 0;
    // }

    & > * {
      vertical-align: top;
      padding: 12px;

      // &:first-child {
      //   position: sticky;
      //   left: 0;
      // }
    }

    &--inline {
      margin-bottom: 8px;
      display: flex;
      align-items: center;
      gap: 4px;
    }

    h5 {
      display: flex;
      align-items: center;
      gap: 4px;
      font-size: 1rem;
    }
  }

  &__header {
    & > * {
      vertical-align: top;
      padding: 12px;
    }

    h2 {
      margin: 0;
      display: flex;
      align-items: center;
      justify-content: center;
      gap: 4px;

      &.left {
        justify-content: flex-start;
      }
    }
  }

  &__value {
    border: 1px solid rgba(200, 200, 200, 0.55);
    border-radius: 8px;
    margin-bottom: 4px;
  }

  &__media-label {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    max-width: 200px;
    display: flex;
    align-items: center;
    justify-content: center;
    gap: 4px;
  }

  &__add-theme,
  &__add-design-token {
    vertical-align: middle;
    margin-left: 8px;
  }

  &__name {
    margin: 0 0 8px;
  }

  &__desc {
    margin-bottom: 8px;
  }

  &__theme-value {
    text-align: center;
  }

  &__add-value {
    margin-top: 4px;
  }

  &__aliasTokens {
    list-style: none;
    padding: 12px;
    margin: 0;

    pre {
      margin: 0;
    }
  }

  &__add-aliasTokens {
    text-align: center;
  }

  &__select-aliasTokens {
    text-align: center;
    padding-top: 8px;
  }
}

a,
a:focus,
a:active {
  text-decoration: none;
  color: inherit;
}
