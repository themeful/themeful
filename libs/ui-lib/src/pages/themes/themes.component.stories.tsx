import { h } from '@stencil/core'
import { action } from '@storybook/addon-actions'
import { of } from 'rxjs'
import sample from './themes.sample.json'

export default {
  title: 'Pages/Themes',
  args: {
    themeBundle$: of(sample),
    onAction: action('triggered action'),
  },
}

export const themes = (args) => {
  return (
    <div>
      <div class="header" style={{ marginBottom: '16px' }}>
        <h1>Themes</h1>
      </div>
      <div class="content">
        <div style={{ position: 'relative' }}>
          <tf-themes {...args} />
        </div>
      </div>
    </div>
  )
}
