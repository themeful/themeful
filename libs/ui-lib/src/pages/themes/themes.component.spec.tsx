import { newSpecPage } from '@stencil/core/testing'
import { ThemesComponent } from './themes.component'

describe('ThemesComponent', () => {
  it('renders', async () => {
    const { root } = await newSpecPage({
      components: [ThemesComponent],
      html: '<tf-themes></tf-themes>',
    })
    expect(root).toMatchSnapshot()
  })

  it('renders with values', async () => {
    const { root } = await newSpecPage({
      components: [ThemesComponent],
      html: `<tf-themes></tf-themes>`,
    })
    expect(root).toMatchSnapshot()
  })
})
