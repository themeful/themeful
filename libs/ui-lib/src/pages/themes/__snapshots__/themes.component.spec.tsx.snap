// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemesComponent renders 1`] = `
<tf-themes>
  <mock:shadow-root>
    <stencil-route-title pagetitle="Themeful - Themes"></stencil-route-title>
    <tf-navigation size="small"></tf-navigation>
    <div class="design-tokens__wrapper">
      <table class="design-tokens">
        <tbody></tbody>
      </table>
    </div>
    <tf-form-integration></tf-form-integration>
  </mock:shadow-root>
</tf-themes>
`;

exports[`ThemesComponent renders with values 1`] = `
<tf-themes>
  <mock:shadow-root>
    <stencil-route-title pagetitle="Themeful - Themes"></stencil-route-title>
    <tf-navigation size="small"></tf-navigation>
    <div class="design-tokens__wrapper">
      <table class="design-tokens">
        <tbody></tbody>
      </table>
    </div>
    <tf-form-integration></tf-form-integration>
  </mock:shadow-root>
</tf-themes>
`;
